### PR TITLE
[RHPAM-2175] Use only adminUser

### PR DESCRIPTION
**Pull request by ruromero**
Signed-off-by: Ruben Romero Montes <rromerom@redhat.com>

Fix https://issues.redhat.com/browse/RHPAM-2175

### CODE REVIEW

File: pkg/apis/app/v2/kieapp_types.go [hunk 1]
```diff
@@ -450,9 +450,12 @@ type CommonConfig struct {
 	DBPassword         string `json:"dbPassword,omitempty"`
 	AMQPassword        string `json:"amqPassword,omitempty"`
 	AMQClusterPassword string `json:"amqClusterPassword,omitempty"`
+	// Deprecated - Remove after 7.6.1 is also removed
```
Comment (spolti) [inline pkg/apis/app/v2/kieapp_types.go]: we will not make 7.6.1, can it be removed now?

Comment (tchughesiv) [reply pkg/apis/app/v2/kieapp_types.go]: I'd imagine this needs to stay until 7.6.0 rolls off. We still support both 7.6 and 7.5.1 installs with the 7.7 operator.

File: deploy/crds/kieapp.crd.yaml [hunk 2]
```diff
@@ -770,15 +770,6 @@ spec:
...
                       description: The password to use for amq cluster user.
-                    controllerPassword:
-                      type: string
-                      description: The password to use for the controllerUser.
-                    serverPassword:
-                      type: string
-                      description: The password to use for the executionUser.
-                    mavenPassword:
-                      type: string
-                      description: The password to use for the mavenUser.
```
Comment (tchughesiv) [inline deploy/crds/kieapp.crd.yaml]: @ruromero @bmozaffa 
More a conversation starter than anything, should we version our api for adds/removes or only "changes" to existing components?

Comment (bmozaffa) [reply deploy/crds/kieapp.crd.yaml]: I think we should decide case by case, with a major driver of our decision being the difficulty and complication of version increments, which I hope will get better. In fact the level of difficulty should even affect our decision to make CRD changes.

As long as CRD versioning remains as difficult as it has been, I would advocate to only version when there is a change, or required field is removed, or similar. New critical and required fields may be an exception, if we want to make sure they are exposed in the CRD.

Comment (ruromero) [reply deploy/crds/kieapp.crd.yaml]: In this case the remove fields are not required and most of the times generated. I don't consider this a breaking change but that's something to always have in mind and agree on beforehand.
Thanks

File: pkg/apis/app/v2/kieapp_types.go (hunk 1 above)
Comment (ruromero) [reply pkg/apis/app/v2/kieapp_types.go]: @spolti I don't want to remove it until version 7.7 is the last one supported by the operator to allow users still deploy v1 of the CRD.

File: pkg/apis/app/v2/kieapp_types.go [hunk 3]
```diff
@@ -450,9 +450,12 @@ type CommonConfig struct {
 	DBPassword         string `json:"dbPassword,omitempty"`
 	AMQPassword        string `json:"amqPassword,omitempty"`
 	AMQClusterPassword string `json:"amqClusterPassword,omitempty"`
+	// Deprecated - Remove when 7.7.0 is the oldest version supported
```
Comment (ruromero) [inline pkg/apis/app/v2/kieapp_types.go]: @spolti I think this comment is clearer

File: pkg/controller/kieapp/defaults/defaults.go [hunk 4]
```diff
@@ -636,10 +636,7 @@ func setPasswords(cr *api.KieApp, isTrialEnv bool) {
 		&cr.Spec.CommonConfig.AdminPassword,
 		&cr.Spec.CommonConfig.DBPassword,
 		&cr.Spec.CommonConfig.AMQPassword,
-		&cr.Spec.CommonConfig.AMQClusterPassword,
-		&cr.Spec.CommonConfig.ControllerPassword,
-		&cr.Spec.CommonConfig.MavenPassword,
-		&cr.Spec.CommonConfig.ServerPassword}
```
Comment (tchughesiv) [inline pkg/controller/kieapp/defaults/defaults.go]: do we need to add some logic here that still adds these passwords if the version is less than 7.7.0? I assume we'll still need these passwords for older product versions?

maybe something like
```shell
	if getMinorImageVersion(cr.Spec.Version) < "77" {
               passwords = append(passwords, &cr.Spec.CommonConfig.ControllerPassword) ...
....
        }
```
???

Comment (ruromero) [reply pkg/controller/kieapp/defaults/defaults.go]: The templates for 7.5.1 and 7.6.0 have the variables set so I think we can just delete them once the oldest templates are 7.7.0

Comment (tchughesiv) [reply pkg/controller/kieapp/defaults/defaults.go]: i think that makes sense ya... so, for this change then, would you add them back to this slice or?

Comment (ruromero) [reply pkg/controller/kieapp/defaults/defaults.go]: Of course, I totally forgot.

**Pull request merged**